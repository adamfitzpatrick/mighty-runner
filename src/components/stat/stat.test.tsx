import { h } from 'preact'
import { shallow } from 'preact-render-spy'
import Stat from '.'

describe('stat component', () => {
  describe('when not provided with a "round" prop', () => {
    it('should render with a specified label and value', () => {
      const stat = shallow(<Stat label='label' value='value' />)
      expect(stat).toMatchSnapshot()
    })
  })

  describe('when provided with a "round" prop', () => {
    it('should render the specified label and value with round styles', () => {
      const stat = shallow(<Stat label='label' value='value' round />)
      expect(stat).toMatchSnapshot()
    })
  })
})
