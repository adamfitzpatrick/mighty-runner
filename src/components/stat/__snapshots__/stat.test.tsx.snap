// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stat component when not provided with a "round" prop should render with a specified label and value 1`] = `
preact-render-spy (1 nodes)
-------
<div class="stat">
  <div class="label">label</div>
  <div>value</div>
</div>

`;

exports[`stat component when provided with a "round" prop should render the specified label and value with round styles 1`] = `
preact-render-spy (1 nodes)
-------
<div class="stat round">
  <div>value</div>
  <div class="label">label</div>
</div>

`;
